src/lap_stats/lap_stats.c

```diff
@@ -651,11 +651,13 @@ static void lapstats_setup(const GpsSnapshot *gps_snapshot)
         const Track *track = NULL;
         track_status_t track_status;
         if (auto_detect_track) {
-                track_status = TRACK_STATUS_AUTO_DETECTED;
-                track = auto_configure_track(NULL, gp);
+                const Track *default_track = (get_tracks()->count == 0) ? &trackConfig->track : NULL;
+                track = auto_configure_track(default_track, gp);
+                track_status = default_track->trackId ? TRACK_STATUS_AUTO_DETECTED : TRACK_STATUS_WAITING_TO_CONFIG;
```

Comment:
This code will result in a SEGFAULT if count is != 0 since default_track will be null.